lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -0,0 +1,1021 @@
+package drivers
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/canonical/lxd/lxd/db"
+	"github.com/canonical/lxd/lxd/locking"
+	"github.com/canonical/lxd/lxd/revert"
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+)
+
+// powerFlexBlockVolSuffix suffix used for block content type volumes.
+const powerFlexBlockVolSuffix = ".block"
+
+// powerFlexISOVolSuffix suffix used for iso content type volumes.
+const powerFlexISOVolSuffix = ".iso"
+
+// powerFlexVolTypePrefixes maps volume type to storage volume name prefix.
+var powerFlexVolTypePrefixes = map[VolumeType]string{
+	VolumeTypeContainer: db.StoragePoolVolumeTypeNameContainer,
+	VolumeTypeVM:        db.StoragePoolVolumeTypeNameVM,
+	VolumeTypeImage:     db.StoragePoolVolumeTypeNameImage,
+	VolumeTypeCustom:    db.StoragePoolVolumeTypeNameCustom,
+}
+
+// powerFlexError contains arbitrary error responses from PowerFlex.
+type powerFlexError map[string]any
+
+// tokenFactory holds an access token together with a read write mutex.
+type tokenFactory struct {
```

Comment:
This type doesnt appear to be powerflex specific, should it be renamed to be so or moved out of the driver?

Reply:
I will move it under `lxd/util/token.go`.